### ISSUE TITLE
docs: Improve examples install instructions

### ISSUE THREAD

**Issue by rht** [projectmesa/mesa#1274]
If you look at https://github.com/projectmesa/mesa/graphs/traffic, #315 is the 3rd most visited page. This means that the readme of the examples are not clear enough to indicate that they must be in the directory containing the requirements.txt.

**Comment by rht** [comment]
#1275 is a good example on how to clarify the readme. Help wanted for the rest of the examples!

**Comment by Corvince** [comment]
I don't think #1275 is a good way to solve this problem. I'll explain myself later (probably tomorrow) but please don't change the other examples before having a discussion

**Comment by tpike3** [comment]
@Corvince intriguing, I look forward to seeing what you are thinking.

**Comment by Corvince** [comment]
Yes, sorry for not stating my concerns directly, but I was on the move and just saw that recently there have been a lot of contributions to mesa, which is great, but I feared someone would start working on this directly. And I also wanted to commit myself ;)

Anyways, I have 2 main concerns with instructing people to use git to clone the repository: 
  1. `git` is not always available. If people trip over `pip install -r requirements` (which I think I also did using mesa for the first time), they are probably rather inexperienced. People without a git installation (i.e. primarly windows users) will probably trip over a failing `git clone` as well.
  2. If people manage to clone the repository, they now have a local, but uninstalled copy of mesa. Chances are they will try to modify mesa code and wonder why it doesn't work, not realizing this is not the mesa code they use when doing `import mesa`.

I am not sure what the best intermediant solution is, but the cleanest solution I always had on top of my mind was to create a `mesa download_examples` command that downloads and extracts a zip of the matching examples. But we would need to provide that versioned zip. We could automate this with each release and for example use Github Pages, so we have a link like `https://projectmesa.github.io/examples/v0.9.0/examples.zip`. Versioning is also important because currently the examples might be out of sync with the latest or certainly older mesa releases and even if people manage to clone them might not be able to run them.

**Comment by rht** [comment]
#1228 is another option.

**Comment by Corvince** [comment]
> #1275 may not be a comprehensive solution to the problem, but at least it makes the steps explicit, instead of assuming that people have already cloned the Git repo. People will see the `git clone` command and look up on the internet on how to set up Git. They no longer assume: "oh, `pip install mesa` is sufficient!".

Hmm I think we are getting somewhere, where we make too many assumptions on what people might or might not think. But I don't follow the argument why people would think `pip install mesa` is sufficient when it fails to install the requirements vs when git is not installed. 

But I totally agree that #1275 is an intermediant solution. In that case however I would prefer to also state the pitfalls explicitly.

It is also not surprising that people mention git cloning, since this is currently required. Doesn't mean its a good solution. Case in point: If you clone the repo right now and try to run the `sugarscape_cg` model it will fail with the latest release version (mesa 0.9.0).

Another example for _executing_ an example is deploying it on Heroku: [PR](https://github.com/projectmesa/mesa-schelling-example/pull/1) and example: https://mesa-schelling-example.herokuapp.com/ 

#1228 has the same discussion point as #568, namely whether the examples should be a part of mesa or inside their own repo (#134)

**Comment by rht** [comment]
>  But I don't follow the argument why people would think pip install mesa is sufficient when it fails to install the requirements vs when git is not installed.

Here are some data points:
- "However, I immediately ran into a problem, after using pip install mesa, I tried to install the requirements (pip install -r requirements.txt) in the terminal." https://github.com/projectmesa/mesa/issues/315#issue-179973809=
- https://github.com/projectmesa/mesa/issues/315#issuecomment-672608402=

> If you clone the repo right now and try to run the sugarscape_cg model it will fail with the latest release version (mesa 0.9.0).

This is a bug that needs to be addressed on its own. I don't recall any breaking changes between latest main and latest release (0.9.0), so something must have slipped through.

I'd say the scope of this issue (#1274) should be limited to improving/clarifying the examples *install instruction*, and not more.

We should discuss the general strategy on making it easy to run examples via Heroku demo, `import mesa.examples` #1228 / `$ mesa examples` #568, `mesa download_examples`, Jupyter GUI #1263, and so in, in a separate space.

**Comment by Corvince** [comment]
> > If you clone the repo right now and try to run the sugarscape_cg model it will fail with the latest release version (mesa 0.9.0).
> 
> This is a bug that needs to be addressed on its own. I don't recall any breaking changes between latest main and latest release (0.9.0), so something must have slipped through.

Don't worry you don't need to recall anything here. This was just an example. I also wouldn't consider it a bug, the state of mesa and the examples is consistent in itself. This is a general problem with the current structure. Try to install any older version of mesa and run the examples, a lot of them probably won't work. 


Just to be clear, I agree that #1275 is an immediate solution to this problem. And if someone has the time and energy to apply it to all examples feel free to do so - but it will be a change that should be reverted some time in the future when we have a better running-the-examples strategy. I am just not sure if it is worth the effort. (And I bet someone will eventually ask why the `git clone` command fails)

**Comment by rht** [comment]
> Try to install any older version of mesa and run the examples, a lot of them probably won't work.

OK, this can be mitigated by doing `mesa>=<last major version>` in the requirements.txt of the example, and by explicitly stating in the readme that the user must run the `pip install -r requirements.txt` when they want to run the example for the first time.

I see #1275 as a complementary solution to `mesa examples` / `mesa ...` for people who prefer to see the source code from a `git clone`, and so, should stay. Some people may find the `mesa ...` interface to be opaque. Again, explicit instruction is the key here.

**Comment by rht** [comment]
> But this is the other way around. This forces the mesa version to match the example. The usual need is to find examples matching the used version. Also, this obviously doesn't work for unreleased versions of mesa, i.e. changes in the main branch.

If you want examples that work on `mesa==<old release>`, you can do `git checkout <old release>`.

> It depends on what solution we will come up with, but mesa download_examples would of course download the examples source code, I don't understand what is opaque about that?

Because some people (not all) prefer to see the source code, the exact commit hash, *and the past revisions* from Git. It's more transparent.

**Comment by ghost** [comment]
Hi! I’m new to Mesa and would like to help with this issue. I’ll look at the existing examples and improve the install instructions where needed.

**Comment by DashamiJituri** [comment]
Hi! I’m a new contributor and I’d like to help improve the install instructions for the #1274 remaining example READMEs.
I’ve recently worked on clarifying run instructions for a couple of examples and would be happy to extend similar improvements to others if that would be helpful.

**Comment by B2prakash** [comment]
Hi, I'd like to work on this. I just set up Mesa locally and went through the install process myself — I have fresh perspective on what instructions beginners need. Happy to submit a PR.

**Comment by Artemiz0307** [comment]
Hi, I'm new to Mesa and preparing for GSoC. 
I'd like to work on improving the installation instructions for the examples. 
Could I take this issue?

**Comment by Artemiz0307** [comment]
Hi, I would like to work on improving the installation instructions in the examples as described in this issue. Could you please assign this to me?

**Comment by quaquel** [comment]
A lot has changed on the docs since this issue was opened. Before opening any PR related to this, Let's first assess what the current situation is. What is good, and what might be improved?

From my perspective, I am inclined to move the installation instructions out of the example-level readme and into the readme at the root of the examples folder. The main thing to explain here is that the run command has to be run from the specific example folder For installation instructions, it might make sense to refer back to the installation instructions at the landing page. That is, let's avoid having installation instructions spread across the docs but refer back where relevant to a single location.

**Comment by prakritiie** [comment]
Hi! I updated the Schelling example README to reference the central installation instructions and clarify that the run command must be executed from the example folder.

Before opening a new pull request, I wanted to confirm that this approach aligns with the current documentation direction discussed here. I'm happy to adjust based on feedback.

**Comment by quaquel** [comment]
I don't want to have PRs for each individual example, but can you share the relevant part of the updated readme here?

If we agree, we can just implement this change in all Readme's in a single PR.

**Comment by prakritiie** [comment]
Sure! Here is the updated README that references the central installation instructions.

## How to Run

First, install Mesa by following the installation instructions in the main documentation:

See the installation guide in the main README:
https://github.com/projectmesa/mesa#using-mesa

Then navigate to this example directory and run the application:

```bash
solara run app.py
```

Note: The command must be executed from this example folder.

Open the displayed local URL in your browser.



Please let me know if this structure looks good, and I'm happy to help apply it consistently across the example READMEs.
